From 2495fa0c0591589a02506f494c5be6067127f674 Mon Sep 17 00:00:00 2001
From: William Wu <william.wu@rock-chips.com>
Date: Thu, 19 Nov 2020 16:47:15 +0800
Subject: [PATCH] arm64: dts: rockchip: fix usb nodes for rk356x boards

This patch adds phy property for usb on various rk356x boards.
3566 EVB1 1 x USB2.0 OTG(Micro USB2.0) + 1 x USB3.0 HOST + 2 x USB2.0 HOST
3566 EVB2 1 x USB2.0 OTG(Micro USB2.0) + 1 x USB3.0 HOST + 2 x USB2.0 HOST
3566 EVB3 1 x USB2.0 OTG(Micro USB2.0) + 1 x USB3.0 HOST(mux with SATA, default for SATA) + 2 x USB2.0 HOST
3568 EVB1 1 x USB3.0 OTG(Type-A USB3.0 + Micro USB2.0)+ 1 x USB3.0 HOST + 2 x USB2.0 HOST
3568 EVB6 1 x USB3.0 OTG(Type-A USB3.0 + Micro USB2.0)+ 3 x USB2.0 HOST

Change-Id: If1e2cdb03e50e770337648f59f0375034b7062cd
Signed-off-by: William Wu <william.wu@rock-chips.com>
---
 .../boot/dts/rockchip/rk3566-evb1-ddr4-v10.dtsi | 13 +++++++++++++
 .../boot/dts/rockchip/rk3566-evb2-lp4x-v10.dtsi | 17 +++++++++++++++++
 .../boot/dts/rockchip/rk3566-evb3-ddr3-v10.dtsi | 17 +++++++++++++++++
 .../boot/dts/rockchip/rk3566-rk817-tablet.dts   |  2 ++
 .../boot/dts/rockchip/rk3568-evb1-ddr4-v10.dtsi |  8 ++++++++
 .../boot/dts/rockchip/rk3568-evb6-ddr3-v10.dtsi | 10 ++++++++++
 .../rockchip/rk3568-iotest-ddr3-v10-linux.dts   |  2 ++
 arch/arm64/boot/dts/rockchip/rk3568-nvr.dtsi    |  4 ++++
 8 files changed, 73 insertions(+)

diff --git a/arch/arm64/boot/dts/rockchip/rk3566-evb1-ddr4-v10.dtsi b/arch/arm64/boot/dts/rockchip/rk3566-evb1-ddr4-v10.dtsi
index 4943a9253431..816c727ab2dc 100644
--- a/arch/arm64/boot/dts/rockchip/rk3566-evb1-ddr4-v10.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3566-evb1-ddr4-v10.dtsi
@@ -193,6 +193,19 @@
 	pinctrl-0 = <&uart1m1_xfer &uart1ctsnm1>;
 };
 
+&usbdrd_dwc3 {
+	dr_mode = "otg";
+	phys = <&u2phy0_otg>;
+	phy-names = "usb2-phy";
+	extcon = <&usb2phy0>;
+	maximum-speed = "high-speed";
+	status = "okay";
+};
+
+&usbdrd30 {
+	status = "okay";
+};
+
 &vcc3v3_lcd0_n {
 	gpio = <&gpio0 RK_PC0 GPIO_ACTIVE_HIGH>;
 	enable-active-high;
diff --git a/arch/arm64/boot/dts/rockchip/rk3566-evb2-lp4x-v10.dtsi b/arch/arm64/boot/dts/rockchip/rk3566-evb2-lp4x-v10.dtsi
index 1bdf83ec4db0..0ca256af7165 100644
--- a/arch/arm64/boot/dts/rockchip/rk3566-evb2-lp4x-v10.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3566-evb2-lp4x-v10.dtsi
@@ -22,6 +22,10 @@
 	};
 };
 
+&combphy1_usq {
+	status = "okay";
+};
+
 /*
  * mipi_dphy0 needs to be enabled
  * when dsi0 is enabled
@@ -196,6 +200,19 @@
 	pinctrl-0 = <&uart1m0_xfer &uart1ctsnm0>;
 };
 
+&usbdrd_dwc3 {
+	dr_mode = "otg";
+	phys = <&u2phy0_otg>;
+	phy-names = "usb2-phy";
+	extcon = <&usb2phy0>;
+	maximum-speed = "high-speed";
+	status = "okay";
+};
+
+&usbdrd30 {
+	status = "okay";
+};
+
 &vcc3v3_lcd0_n {
 	gpio = <&gpio0 RK_PC7 GPIO_ACTIVE_HIGH>;
 	enable-active-high;
diff --git a/arch/arm64/boot/dts/rockchip/rk3566-evb3-ddr3-v10.dtsi b/arch/arm64/boot/dts/rockchip/rk3566-evb3-ddr3-v10.dtsi
index 1ee6daf7bd96..61238adaf5e5 100644
--- a/arch/arm64/boot/dts/rockchip/rk3566-evb3-ddr3-v10.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3566-evb3-ddr3-v10.dtsi
@@ -22,6 +22,10 @@
 	};
 };
 
+&combphy1_usq {
+	status = "okay";
+};
+
 /*
  * mipi_dphy0 needs to be enabled
  * when dsi0 is enabled
@@ -251,6 +255,19 @@
 	status = "okay";
 };
 
+&usbdrd_dwc3 {
+	dr_mode = "otg";
+	phys = <&u2phy0_otg>;
+	phy-names = "usb2-phy";
+	extcon = <&usb2phy0>;
+	maximum-speed = "high-speed";
+	status = "okay";
+};
+
+&usbdrd30 {
+	status = "okay";
+};
+
 &vcc3v3_lcd0_n {
 	gpio = <&gpio0 RK_PC0 GPIO_ACTIVE_HIGH>;
 	enable-active-high;
diff --git a/arch/arm64/boot/dts/rockchip/rk3566-rk817-tablet.dts b/arch/arm64/boot/dts/rockchip/rk3566-rk817-tablet.dts
index 02ed1018c830..bba3b1866e9b 100644
--- a/arch/arm64/boot/dts/rockchip/rk3566-rk817-tablet.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3566-rk817-tablet.dts
@@ -1117,6 +1117,8 @@
 
 &usbdrd_dwc3 {
 	dr_mode = "otg";
+	phys = <&u2phy0_otg>;
+	phy-names = "usb2-phy";
 	extcon = <&usb2phy0>;
 	maximum-speed = "high-speed";
 	status = "okay";
diff --git a/arch/arm64/boot/dts/rockchip/rk3568-evb1-ddr4-v10.dtsi b/arch/arm64/boot/dts/rockchip/rk3568-evb1-ddr4-v10.dtsi
index 14e68792e22d..3956d84a69f1 100644
--- a/arch/arm64/boot/dts/rockchip/rk3568-evb1-ddr4-v10.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3568-evb1-ddr4-v10.dtsi
@@ -74,6 +74,14 @@
 	};
 };
 
+&combphy0_us {
+	status = "okay";
+};
+
+&combphy1_usq {
+	status = "okay";
+};
+
 &combphy2_psq {
 	status = "okay";
 };
diff --git a/arch/arm64/boot/dts/rockchip/rk3568-evb6-ddr3-v10.dtsi b/arch/arm64/boot/dts/rockchip/rk3568-evb6-ddr3-v10.dtsi
index 59929c20e58a..33a8e79e7293 100644
--- a/arch/arm64/boot/dts/rockchip/rk3568-evb6-ddr3-v10.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3568-evb6-ddr3-v10.dtsi
@@ -62,6 +62,10 @@
 	};
 };
 
+&combphy0_us {
+	status = "okay";
+};
+
 &combphy2_psq {
 	status = "disabled";
 };
@@ -212,6 +216,12 @@
 	pinctrl-0 = <&uart1m0_xfer &uart1ctsnm0>;
 };
 
+&usbhost_dwc3 {
+	phys = <&u2phy0_host>;
+	phy-names = "usb2-phy";
+	status = "okay";
+};
+
 &vcc3v3_lcd0_n {
 	gpio = <&gpio0 RK_PC7 GPIO_ACTIVE_HIGH>;
 	enable-active-high;
diff --git a/arch/arm64/boot/dts/rockchip/rk3568-iotest-ddr3-v10-linux.dts b/arch/arm64/boot/dts/rockchip/rk3568-iotest-ddr3-v10-linux.dts
index dbad77cf9f26..3f09dc5d52c2 100644
--- a/arch/arm64/boot/dts/rockchip/rk3568-iotest-ddr3-v10-linux.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3568-iotest-ddr3-v10-linux.dts
@@ -55,6 +55,8 @@
 
 &usbdrd_dwc3 {
 	dr_mode = "otg";
+	phys = <&u2phy0_otg>;
+	phy-names = "usb2-phy";
 	extcon = <&usb2phy0>;
 	maximum-speed = "high-speed";
 	status = "okay";
diff --git a/arch/arm64/boot/dts/rockchip/rk3568-nvr.dtsi b/arch/arm64/boot/dts/rockchip/rk3568-nvr.dtsi
index 53d13f183aae..543d6f1e59bb 100644
--- a/arch/arm64/boot/dts/rockchip/rk3568-nvr.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3568-nvr.dtsi
@@ -378,6 +378,8 @@
 
 &usbdrd_dwc3 {
 	dr_mode = "peripheral";
+	phys = <&u2phy0_otg>;
+	phy-names = "usb2-phy";
 	maximum-speed = "high-speed";
 	status = "okay";
 };
@@ -387,6 +389,8 @@
 };
 
 &usbhost_dwc3 {
+	phys = <&u2phy0_host>;
+	phy-names = "usb2-phy";
 	status = "okay";
 };
 
-- 
2.35.3

